charms/kserve-controller/src/charm.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
+
+"""Charm the service.
+
+Refer to the following post for a quick-start guide that will help you
+develop a new k8s charm using the Operator Framework:
+
+    https://discourse.charmhub.io/t/4208
+"""
+
+import logging
+import traceback
+
+from charmed_kubeflow_chisme.kubernetes import KubernetesResourceHandler
+from charmed_kubeflow_chisme.lightkube.batch import delete_many
+from charmed_kubeflow_chisme.exceptions import ErrorWithStatus
+from charmed_kubeflow_chisme.pebble import update_layer
+from lightkube.generic_resource import load_in_cluster_generic_resources
+from lightkube import ApiError
+from ops.charm import CharmBase
+from ops.main import main
+from subprocess import check_call
+from pathlib import Path
+from base64 import b64encode
+
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
+from ops.pebble import ChangeError, Layer
+
+#from lightkube_custom_resources.serving import ClusterServingRuntime_v1alpha1
+
+log = logging.getLogger(__name__)
+
+K8S_RESOURCE_FILES = [
+    "src/templates/crd_manifests.yaml.j2",
+    "src/templates/auth_manifests.yaml.j2",
+    "src/templates/serving_runtimes_manifests.yaml.j2",
+    "src/templates/webhook_manifests.yaml.j2",
+    "src/templates/configmap_manifests.yaml.j2",
+]
+CONFIG_FILE = "src/config/inference-config.j2"
+
+
+class CheckFailed(Exception):
```

Comment:
There is chisme ErrorWithStatus that can be used instead.

Reply:
True, I'll remove it.

Reply:
[Removed](https://github.com/canonical/kserve-operators/blob/KF-803-kserve-refactor/charms/kserve-controller/src/charm.py)